samples/qa-virtuallistnative/src/views/MainPanel.js

```diff
@@ -1,116 +1,93 @@
+	}, [changeData]);
 
-		this.props.changeData(200, false);
-	}
+	const handleChange = useCallback(({value: newValue}) => setValue(newValue), []);
 
-	handleChange = ({value}) => this.setState({value});
+	const onChangeDataSize = useCallback(() => {
+		const dataSize = parseInt(value) || 0;
+		changeData(dataSize, isDisabled);
+	}, [changeData, isDisabled, value]);
 
-	onChangeDataSize = () => {
-		const dataSize = parseInt(this.state.value) || 0;
-		this.props.changeData(dataSize, this.state.isDisabled);
-	};
+	const onToggleChildProps = useCallback(() => {
+		setHasChildProps(!hasChildProps);
+	}, [hasChildProps]);
 
-	onToggleChildProps = () => {
-		this.setState((state) => ({hasChildProps: !state.hasChildProps}));
-	};
+	const onToggleDisabled = useCallback(() => {
+		changeData(listItems.length, !isDisabled);
+		return setIsDisabled(!isDisabled);
+	}, [changeData, isDisabled, listItems.length]);
 
-	onToggleDisabled = () => {
-		this.setState((state, props) => {
-			this.props.changeData(props.listItems.length, !state.isDisabled);
-			return {isDisabled: !state.isDisabled};
-		});
-	};
-
-	renderItem = ({index, text, ...rest}) => {
+	const renderItem = useCallback(({index, text, ...props}) => {
 		return (
-			<ListItem {...rest} index={index}>
-				{this.props.listItems[index].content + (text || '')}
+			<ListItem {...props} index={index}>
+				{listItems[index].content + (text || '')}
 			</ListItem>
 		);
-	};
-
-	render () {
-		const {listItems, ...rest} = this.props;
-
-		delete rest.changeData;
-
-		return (
-			<Panel {...rest}>
-				<Header
-					title="VirtualList Native"
-					type="compact"
-				/>
-				<div className={css.header}>
-					<div>
-						DataSize:
-						<Input
-							onChange={this.handleChange}
-							placeholder={`${listItems.length}`}
-							size="small"
-							style={{width: '5em'}}
-							type="number"
-							value={this.state.value}
-						/>
-						<Button size="small" onClick={this.onChangeDataSize}>Set DataSize</Button>
-						<ToggleButton size="small" onToggle={this.onToggleDisabled}>Disabled Items</ToggleButton>
-						<ToggleButton size="small" onToggle={this.onToggleChildProps}>Child Props</ToggleButton>
-						<LocaleSwitch size="small" />
-						<Heading showLine />
-					</div>
-					<div className={css.list}>
-						<VirtualList
-							childProps={this.state.hasChildProps ? childProps : null}
-							dataSize={listItems.length}
-							focusableScrollbar
-							itemRenderer={this.renderItem}
-							itemSize={ri.scale(60 + 3)}
-						/>
-					</div>
+	}, [listItems]);
+
+	return (
+		<Panel {...rest}>
+			<Header
+				title="VirtualList Native"
+				type="compact"
+			/>
+			<div className={css.header}>
+				<div>
+					DataSize:
+					<Input
+						onChange={handleChange}
+						placeholder={`${listItems.length}`}
+						size="small"
+						style={{width: '5em'}}
+						type="number"
+						value={value}
+					/>
+					<Button size="small" onClick={onChangeDataSize}>Set DataSize</Button>
```

Comment:
Also here, sort props please and also for lines 74 and 75

Reply:
Done